tools/ports/libmodplug.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2021 The Emscripten Authors.  All rights reserved.
+# Emscripten is available under two separate licenses, the MIT license and the
+# University of Illinois/NCSA Open Source License.  Both these licenses can be
+# found in the LICENSE file.
+
+import os
+import shutil
+import logging
+from tools import building
+
+TAG = '11022021'
+HASH = 'f770031ad6c2152cbed8c8eab8edf2be1d27f9e74bc255a9930c17019944ee5fdda5308ea992c66a78af9fe1d8dca090f6c956910ce323f8728247c10e44036b'
+
+
+def needed(settings):
+  return settings.USE_MODPLUG
+
+
+def get(ports, settings, shared):
+  ports.fetch_project('libmodplug', 'https://github.com/jancc/libmodplug/archive/v' + TAG + '.zip', 'libmodplug-' + TAG, sha512hash=HASH)
+
+  def create(final):
+    logging.info('building port: libmodplug')
+    ports.clear_project_build('libmodplug')
+
+    source_path = os.path.join(ports.get_dir(), 'libmodplug', 'libmodplug-' + TAG)
+    dest_path = os.path.join(ports.get_build_dir(), 'libmodplug')
+
+    configure_args = [
+      'cmake',
+      '-G', 'Unix Makefiles',
+      '-B' + dest_path,
+      '-H' + source_path,
+      '-DCMAKE_BUILD_TYPE=Release',
+      '-DCMAKE_INSTALL_PREFIX=' + dest_path,
+    ]
+
+    extra_cflags = []
+
+    if settings.RELOCATABLE:
+      extra_cflags.append('-fPIC')
+
+    if len(extra_cflags):
+      configure_args += ['-DCMAKE_CXX_FLAGS="{}"'.format(' '.join(extra_cflags))]
+      configure_args += ['-DCMAKE_C_FLAGS="{}"'.format(' '.join(extra_cflags))]
+
+    building.configure(configure_args)
```

Comment:
it is better to avoid configure, as that may not work on windows for example. See mpg123.py for an example of how to avoid it.

Reply:
Done! I've changed it to basically the same setup as mpg123, with a preconfigured config.h and explicit compilation of the library's sources.